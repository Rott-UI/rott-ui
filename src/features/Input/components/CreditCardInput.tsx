// React Imports
import {FC} from 'react'

// React Native Imports
import {StyleSheet} from 'react-native'

// Component Imports
import {InputProps} from '../models'
import {InputStyles} from '../styles'

// Package Imports
import {MaskedTextInput} from 'react-native-mask-text'

export const CreditCardInput: FC<Omit<InputProps, 'label' | 'name'>> = ({
  fontSize,
  onChangeText,
  theme,
  disabled,
  size,
  ...props
}) => {
  const handleOnChangeText = (text: string) => {
    const replacedText = text.replace(/[^0-9]/g, '')
    onChangeText!(replacedText)
  }

  return (
    <MaskedTextInput
      editable={!disabled}
      mask='9999-9999-9999-9999'
      placeholder='**** **** **** ****'
      keyboardType='number-pad'
      maxLength={19}
      onChangeText={(text, rawText) => handleOnChangeText(rawText)}
      style={StyleSheet.flatten([InputStyles({fontSize, theme, size}).defaultTextInputStyle])}
      {...props}
    />
  )
}
