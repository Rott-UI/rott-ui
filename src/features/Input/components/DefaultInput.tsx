// React Imports
import {FC} from 'react'

// React Native Imports
import {StyleSheet, TextInput} from 'react-native'

// Component Imports
import {InputProps} from '../models'
import {InputStyles} from '../styles'

export const DefaultInput: FC<Omit<InputProps, 'name'>> = ({
  label,
  placeholder,
  fontSize,
  theme,
  size,
  ...props
}) => {
  return (
    <TextInput
      keyboardType='default'
      autoCapitalize='none'
      placeholder={placeholder ?? (typeof label === 'string' ? label : undefined)}
      style={StyleSheet.flatten([InputStyles({fontSize, theme, size}).defaultTextInputStyle])}
      {...props}
    />
  )
}
