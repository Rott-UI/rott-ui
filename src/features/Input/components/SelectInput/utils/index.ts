export * from './listHeightNormalizer'
export * from './modalHeightPercentageNormalizer'
