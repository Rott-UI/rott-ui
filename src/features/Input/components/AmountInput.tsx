// React Imports
import {FC, useCallback, useEffect, useRef, useState} from 'react'

// React Native Imports
import {StyleSheet, TextInput} from 'react-native'

// Component Imports
import {InputProps} from '../models'
import {InputStyles} from '../styles'
import {Item} from '../../Item'
import {Label} from '../../Label'
import {Icon} from '../../Icon'

// Util and Lib Imports
import {themeConfig} from '../../../providers'

// Package Imports
import {AmountInputStyles} from '../styles/AmountInput.style'
import {InputStyleNormalizer} from '../utils/inputNormalizer'

export const AmountInput: FC<Omit<InputProps, 'placeholder' | 'label' | 'name'>> = ({
  fontSize,
  onChangeText,
  value,
  theme,
  disabled,
  size,
  ...props
}) => {
  const amountRef = useRef<TextInput>(null)
  const [amount, setAmount] = useState('')
  const [currency, setCurrency] = useState('')

  const resetInternalStates = useCallback(() => {
    setAmount('')
    setCurrency('')
    onChangeText!('0.00')
  }, [])

  const replaceTextWithNumberOrEmpty = (text: string) =>
    text.replace(/[^0-9]/g, '').replace(/^0+/, '')

  const amountNormalizer = (text: string) => {
    if (!text) return resetInternalStates()

    const replacedAmount = replaceTextWithNumberOrEmpty(text)
    const reversedText = replacedAmount.split('').reverse().join('') ?? ''
    const dottedText = reversedText?.match(/.{1,3}/g)?.join('.')
    const normalizedText = dottedText?.split('').reverse().join('')

    setAmount(normalizedText ?? '')
  }

  const handleTextChange = (inputTextAmount: string, inputTextCurrency: string) => {
    if (!inputTextAmount && !inputTextCurrency) return

    let amountFormat = replaceTextWithNumberOrEmpty(inputTextAmount)
    let currencyFormat = replaceTextWithNumberOrEmpty(inputTextCurrency)

    amountFormat = amountFormat.length === 0 ? '0' : amountFormat

    currencyFormat = currencyFormat.length === 0 ? '00' : currencyFormat
    currencyFormat = currencyFormat.length === 1 ? `${currencyFormat}0` : currencyFormat

    onChangeText!(`${amountFormat}.${currencyFormat}`)
  }

  const placeholderColorNormalizer =
    theme === 'dark'
      ? themeConfig.colors['white']
      : (!amount && !currency) ||
        ((amount === '0' || amount === '') && (!currency || currency === '' || currency === '00'))
      ? themeConfig.colors['grey-200']
      : themeConfig.colors['grey-900']

  useEffect(() => {
    const initialAmount = value?.split(',')[0]
    const initialCurrency = value?.split(',')[1]

    setAmount(initialAmount ?? '')
    setCurrency(initialCurrency ?? '')
  }, [])

  useEffect(() => {
    if (value === '' || !value || value === '0') resetInternalStates()
  }, [value])

  useEffect(() => {
    handleTextChange(amount, currency)
  }, [amount, currency])

  return (
    <Item relative onTouchStart={() => amountRef.current?.focus()}>
      <Item row {...props}>
        <TextInput
          ref={amountRef}
          nativeID='amount-native-id'
          testID='amount-test-id'
          editable={!disabled}
          placeholder='0'
          maxLength={11}
          value={amount}
          keyboardType='number-pad'
          onChangeText={(text) => amountNormalizer(text)}
          style={StyleSheet.flatten([
            InputStyles({fontSize, theme, size}).defaultTextInputStyle,
            AmountInputStyles().amountInputStyle,
            {
              color: placeholderColorNormalizer,
            },
          ])}
          placeholderTextColor={placeholderColorNormalizer}
        />

        <Item
          justifyContentFlexEnd
          paddingBottom={InputStyleNormalizer({size}).bottomElementPadding}>
          <Label
            fontSize={
              fontSize ??
              InputStyleNormalizer({
                size,
              }).placeholderSize
            }
            fontWeight='bold'
            color={placeholderColorNormalizer as string}>
            ,
          </Label>
        </Item>

        <TextInput
          testID='currency-test-id'
          maxLength={2}
          placeholder='00'
          value={currency}
          keyboardType='number-pad'
          onChangeText={(text) => setCurrency(replaceTextWithNumberOrEmpty(text).substring(0, 2))}
          onFocus={() => (currency === '00' ? setCurrency('') : undefined)}
          onBlur={() => (currency.isEmpty() || currency === '0' ? setCurrency('00') : undefined)}
          style={StyleSheet.flatten([
            InputStyles({fontSize, theme, size}).defaultTextInputStyle,
            AmountInputStyles().amountInputStyle,
            {
              color: placeholderColorNormalizer,
            },
          ])}
          placeholderTextColor={placeholderColorNormalizer}
        />
      </Item>

      <Item absolute right={0} bottom={InputStyleNormalizer({size}).icon.paddingBottom}>
        <Icon
          name='TL'
          width={InputStyleNormalizer({size}).icon.width}
          height={InputStyleNormalizer({size}).icon.height}
          color={themeConfig.colors['grey-200']}
          mode='fill'
          noStroke
        />
      </Item>
    </Item>
  )
}
