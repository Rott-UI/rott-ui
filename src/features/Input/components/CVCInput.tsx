// React Imports
import {FC} from 'react'

// React Native Imports
import {Alert, StyleSheet, TextInput} from 'react-native'

// Component Imports
import {InputProps} from '../models'
import {InputStyles, CVCInputStyles} from '../styles'
import {Item} from '../../Item'
import {Icon} from '../../Icon'
import {Pressable} from '../../Pressable'

// Util and Lib Imports
import {formatMessage} from '../../../utils'

export const CVCInput: FC<Omit<InputProps, 'label' | 'placeholder' | 'name'>> = ({
  fontSize,
  onChangeText,
  theme,
  size,
  ...props
}) => {
  const handleTextChange = (inputText: string) => {
    let text = inputText.replace(/[^0-9]/g, '')
    text = text.length > 3 ? text.substring(0, 3) : text
    onChangeText!(text)
  }

  return (
    <Item row>
      <TextInput
        editable={!props.disabled}
        placeholder='***'
        style={StyleSheet.flatten([InputStyles({fontSize, theme, size}).defaultTextInputStyle])}
        keyboardType='number-pad'
        maxLength={3}
        onChangeText={handleTextChange}
        {...props}
      />

      {/* TODO: CVC Info butonuna kart arkayüzü eklenmeli */}
      <Pressable
        testID='info-icon-test-id'
        style={CVCInputStyles().infoIcon}
        justifyContentCenter
        alignItemsCenter
        onPress={() =>
          Alert.alert(formatMessage('CVC.INFO.TITLE'), formatMessage('CVC.INFO.DESCRIPTION'), [
            {
              text: formatMessage('COMMON.OK'),
              style: 'cancel',
            },
          ])
        }>
        <Icon name='CHECK_CIRCLE' height={24} width={24} variant='grey-900' />
      </Pressable>
    </Item>
  )
}
