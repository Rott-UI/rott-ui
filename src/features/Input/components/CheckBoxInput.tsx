// React Imports
import {FC, ReactNode} from 'react'

// React Native Imports

// Component Imports
import {InputProps} from '../models'
import {Label} from '../../Label'
import {Item} from '../../Item'
import {Pressable} from '../../Pressable'

// Util and Lib Imports
import {InputStyles} from '../styles'
import {InputStyleNormalizer} from '../utils/inputNormalizer'
import {themeConfig} from '../../../providers'

interface CheckBoxProps extends Omit<InputProps, 'placeholder' | 'name' | 'label'> {
  description?: ReactNode
  checked?: boolean
  onCheckChange?: (checked: boolean) => void
}

export const CheckBoxInput: FC<CheckBoxProps> = ({
  description,
  checked,
  fontSize,
  theme,
  size,
  disabled,
  onCheckChange,
}) => {
  return (
    <Pressable
      flex={0}
      testID='checkbox-input-test-id'
      onPress={() => !disabled && onCheckChange!(!checked)}>
      <Item
        row
        alignItemsCenter
        style={InputStyles({theme, size}).defaultTextInputStyle}
        testID='checkbox-container-test-id'>
        <Item
          borderRadius={8}
          borderColor={checked ? themeConfig.colors['primary'] : themeConfig.colors['grey-200']}
          borderWidth={2}
          backgroundColor='white'
          width={24}
          height={24}
          alignItemsCenter
          justifyContentCenter>
          {checked && (
            <Item
              testID='checkbox-checked-test-id'
              borderWidth={2}
              borderColor={themeConfig.colors['neutral-blue-alpha']}
              borderRadius={4}
              backgroundColor={themeConfig.colors['primary']}
              width={15}
              height={15}
            />
          )}
        </Item>

        {description && typeof description === 'string' && (
          <Label
            testID='checkbox-default-label-test-id'
            fontSize={fontSize ?? InputStyleNormalizer({size}).placeholderSize}
            fontWeight='bold'
            marginLeft={12}
            variant={theme === 'dark' ? 'white' : 'grey-900'}>
            {description}
          </Label>
        )}

        {description && typeof description !== 'string' && <>{description}</>}
      </Item>
    </Pressable>
  )
}
