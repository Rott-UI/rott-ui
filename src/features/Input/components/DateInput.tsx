// React Native Imports
// import {Modal} from 'react-native'

// React Imports
import React, {FC, useState} from 'react'

// Component Imports
import {InputProps} from '../models'
import {Item} from '../../Item'
import {Pressable} from '../../Pressable'
import {DateInputStyles} from '../styles'
import {Modal, useModal} from '../../Modal'

// Feature Imports
import {getLanguageState} from '../../../features/app'

// Util and Lib Imports
import {display} from '../../../utils'
import {formatByDateMode} from '../utils'

// Package Imports
import DatePicker from 'react-native-date-picker'
import {COLOURS} from '../../../constants'
import {InputStyleNormalizer} from '../utils/inputNormalizer'
import {useAppSelector} from '../../../hooks'

interface DateInputProps extends Omit<InputProps, 'label' | 'name'> {}

/**
 *
 * DateInput'un kullanımı diğer inputlara göre biraz daha farklı.
 * Yapısı gereği diğer inputlarda kullandığımız "onChangeText" event'i ile değişen değeri almamız mümkün olmadığı için
 * formik'in içinde yer alan "setFieldValue" metodunu kullanmamız gerek.
 *
 * ### **Örnek**:
 * <Input onDateChange={(date) => setFieldValue('date', date)} />
 *
 * @property {(date: Date) => void)} onDateChange: seçilen tarih değerini almak içim bu property'nin kullanılması gerekmektedir.
 * @property {'date' | 'time' | 'datetime'} mode: İstenilen mod neyse ona göre belirtilmelidir. Default değer datetime olarak ayarlanmıştır.
 * @property {boolean} dontAllowClearing: Eğer date input'un değerinin temizlenmesine izin verilmemesi gerekiyorsa bu propert kullanılmalıdır.
 * dontAllowClearing={true} yerine sadece downAllowClearing olarak verirseniz değer otomatik olarak true kabul edilir. False olmadığı durumlarda
 * lütfen ={true} şeklinde tanımlama yapmayınız.
 */
export const DateInput: FC<DateInputProps> = ({
  placeholder = 'Lütfen bir tarih seçiniz',
  onDateChange,
  mode = 'date',
  allowClear = false,
  minimumDate,
  maximumDate,
  value,
  testID,
  theme,
  size,
  ...props
}) => {
  const appLanguage = useAppSelector(getLanguageState)

  const [datePickerValue, setDatePickerValue] = useState<Date>(new Date())

  const handleConfirmPress = () => {
    let validDate: Date
    if (minimumDate && datePickerValue.getTime() < minimumDate?.getTime()) validDate = minimumDate
    else if (maximumDate && datePickerValue.getTime() > maximumDate?.getTime())
      validDate = maximumDate
    else validDate = datePickerValue

    onDateChange!(validDate)
    Modal.hideModal(98)
  }

  const handleClearPress = () => {
    hideModal()
    setDatePickerValue(new Date())
    onDateChange!(null as any)
  }

  const {showModal, hideModal} = useModal(
    {
      id: 98,
      testID: 'date-input-modal',
      visible: true,
      height: 30,
      header: (
        <Item
          justifyContentCenter
          backgroundColor={COLOURS.GREY900}
          height={display.heightPixel(50)}
          style={DateInputStyles().dateInputHeaderStyle}>
          <Pressable
            testID='date-input-confirm-button'
            onPress={() => handleConfirmPress()}
            text='Tamam'
            textVariant='white'
            textSize='xl'
            textWeight={700}
            style={DateInputStyles().confirmButtonStyle}
          />

          {allowClear && value && (
            <Pressable
              testID='date-input-clear-button'
              onPress={handleClearPress}
              text='Temizle'
              textVariant='white'
              textSize='xl'
              textWeight={700}
              style={DateInputStyles().cancelButtonStyle}
            />
          )}
        </Item>
      ),
      children: (
        <Item size='full' alignItemsCenter justifyContentCenter>
          <DatePicker
            testID={testID}
            mode={mode}
            date={datePickerValue}
            textColor={COLOURS.GREY900}
            locale={appLanguage?.name}
            minimumDate={minimumDate}
            maximumDate={maximumDate}
            onDateChange={(date) => setDatePickerValue(date)}
            {...props}
          />
        </Item>
      ),
    },
    [datePickerValue]
  )

  return (
    <Item row>
      <Pressable
        size='full'
        height={InputStyleNormalizer({size}).height}
        testID='date-input-value-container'
        flex={0}
        justifyContentCenter
        textSize='xl'
        text={value ? formatByDateMode(mode, value) : placeholder}
        textStyle={DateInputStyles({theme, value}).pressableTextStyle}
        textVariant={theme === 'dark' ? 'white' : value ? 'grey-900' : 'grey-200'}
        onPress={() => {
          showModal()
        }}
        marginTop='2.5%'
      />
    </Item>
  )
}
