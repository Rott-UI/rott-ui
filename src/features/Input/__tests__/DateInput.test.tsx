// Component Imports
import {DateInput} from '../components'

// Util and Lib Imports
import {act, fireEvent, render, waitFor} from '../../../utils'

describe('Date Input -> Custom Input', () => {
  const dateInputTestId = 'date-input-test-id'
  const dateInputValueContainerTestId = 'date-input-value-container'
  const dateInputConfirmButtonTestId = 'date-input-confirm-button'
  const dateInputClearmButtonTestId = 'date-input-clear-button'
  const dateInputModalTestId = 'date-input-modal'
  const currentDate = new Date()

  it('ilk render anında snapshot ile eşleşmeli', () => {
    const renderedDateInput = render(<DateInput testID={dateInputTestId} date={currentDate} />)

    expect(renderedDateInput).toMatchSnapshot()
  })

  it('date inputa tıklandığında input modal olarak açılmalı', () => {
    const {getByTestId} = render(<DateInput testID={dateInputTestId} date={currentDate} />)

    const dateInputValueContainer = getByTestId(dateInputValueContainerTestId)
    fireEvent.press(dateInputValueContainer)

    const dateInputModal = getByTestId(dateInputModalTestId)
    expect(dateInputModal).toBeVisible()
  })

  it('date input allowClear propertysi almadıysa değer temizleme butonu ekranda gözükmemeli.', () => {
    const {getByTestId, queryByTestId} = render(
      <DateInput testID={dateInputTestId} date={currentDate} />
    )

    const dateInputValueContainer = getByTestId(dateInputValueContainerTestId)
    fireEvent.press(dateInputValueContainer)

    const clearButton = queryByTestId(dateInputClearmButtonTestId)
    expect(clearButton).toBeNull()
  })

  it('date input allowClear propertysi aldıysa Temizle butonuna tıklandığında değer temizlenmeli.', async () => {
    const onDateChangeMock = jest.fn()
    const {getByTestId, queryByTestId} = render(
      <DateInput
        testID={dateInputTestId}
        value={currentDate.toISOString()}
        date={currentDate}
        allowClear
        onDateChange={onDateChangeMock}
      />
    )

    const dateInputValueContainer = getByTestId(dateInputValueContainerTestId)
    await act(() => {
      fireEvent.press(dateInputValueContainer)
    })
    const modalElement = getByTestId(dateInputModalTestId)
    expect(modalElement).toBeVisible()
    await waitFor(() => {
      const dateInput = getByTestId(dateInputTestId)
      expect(dateInput).toBeOnTheScreen()
      expect(dateInput).toHaveProp('date')
    })
    const clearButton = getByTestId(dateInputClearmButtonTestId)
    fireEvent.press(clearButton)

    const modalShouldNotVisible = queryByTestId(dateInputModalTestId)
    expect(modalShouldNotVisible).not.toBeOnTheScreen()
  })

  it('date input mode date olarak renderlanmalı', async () => {
    const {getByTestId} = render(<DateInput testID={dateInputTestId} mode='date' />)

    const dateInputValueContainer = getByTestId(dateInputValueContainerTestId)
    await act(() => {
      fireEvent.press(dateInputValueContainer)
    })

    await waitFor(() => {
      const dateInput = getByTestId(dateInputTestId)
      expect(dateInput).toBeOnTheScreen()
      expect(dateInput).toHaveProp('mode', 'date')
    })
  })

  it('date input mode time olarak renderlanmalı', async () => {
    const {getByTestId} = render(<DateInput testID={dateInputTestId} mode='time' />)

    const dateInputValueContainer = getByTestId(dateInputValueContainerTestId)
    await act(() => {
      fireEvent.press(dateInputValueContainer)
    })

    await waitFor(() => {
      const dateInput = getByTestId(dateInputTestId)
      expect(dateInput).toBeOnTheScreen()
      expect(dateInput).toHaveProp('mode', 'time')
    })
  })

  it('date input mode datetime olarak renderlanmalı', async () => {
    const {getByTestId} = render(<DateInput testID={dateInputTestId} mode='datetime' />)

    const dateInputValueContainer = getByTestId(dateInputValueContainerTestId)
    await act(() => {
      fireEvent.press(dateInputValueContainer)
    })

    await waitFor(() => {
      const dateInput = getByTestId(dateInputTestId)
      expect(dateInput).toBeOnTheScreen()
      expect(dateInput).toHaveProp('mode', 'datetime')
    })
  })

  it('minimum date verildikten sonra daha geçmiş bir gün seçilirse tanımlanan minimum date değer olarak atanmalı.', async () => {
    const onDateChangeMock = jest.fn()
    const {getByTestId} = render(
      <DateInput
        testID={dateInputTestId}
        mode='date'
        date={new Date()}
        minimumDate={currentDate}
        onDateChange={onDateChangeMock}
      />
    )

    const dateInputValueContainer = getByTestId(dateInputValueContainerTestId)
    await act(() => {
      fireEvent.press(dateInputValueContainer)
    })

    const dateInput = getByTestId(dateInputTestId)
    expect(dateInput).toBeOnTheScreen()
    await waitFor(() => {
      fireEvent(dateInput, 'onDateChange', new Date('2023.09.14'))

      setTimeout(() => {}, 150)
    })

    const confirmButton = getByTestId(dateInputConfirmButtonTestId)
    await act(() => {
      fireEvent.press(confirmButton)
    })

    expect(onDateChangeMock).toHaveBeenCalledWith(currentDate)
  })

  it('maximum date verildikten sonra daha ileri bir gün seçilirse tanımlanan maximum date değer olarak atanmalı.', async () => {
    const onDateChangeMock = jest.fn(() => currentDate)
    const {getByTestId} = render(
      <DateInput
        testID={dateInputTestId}
        mode='date'
        date={currentDate}
        maximumDate={currentDate}
        onDateChange={onDateChangeMock}
      />
    )

    const dateInputValueContainer = getByTestId(dateInputValueContainerTestId)
    await act(() => {
      fireEvent.press(dateInputValueContainer)
    })

    await waitFor(() => {
      const dateInput = getByTestId(dateInputTestId)
      expect(dateInput).toBeOnTheScreen()

      fireEvent(dateInput, 'onDateChange', new Date(currentDate.setHours(24)))

      const confirmButton = getByTestId(dateInputConfirmButtonTestId)
      fireEvent.press(confirmButton)
    })

    expect(onDateChangeMock).toHaveBeenCalledWith(currentDate)
  })
})
