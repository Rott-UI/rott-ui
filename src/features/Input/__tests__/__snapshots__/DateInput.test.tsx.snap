// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date Input -> Custom Input ilk render anında snapshot ile eşleşmeli 1`] = `
<View
  style={
    {
      "alignItems": undefined,
      "backgroundColor": undefined,
      "bottom": undefined,
      "flex": undefined,
      "flexDirection": "row",
      "flexGrow": undefined,
      "gap": undefined,
      "height": undefined,
      "justifyContent": undefined,
      "left": undefined,
      "marginBottom": undefined,
      "marginLeft": undefined,
      "marginRight": undefined,
      "marginTop": undefined,
      "maxHeight": undefined,
      "maxWidth": undefined,
      "minHeight": undefined,
      "minWidth": undefined,
      "overflow": undefined,
      "paddingBottom": undefined,
      "paddingHorizontal": undefined,
      "paddingLeft": undefined,
      "paddingRight": undefined,
      "paddingTop": undefined,
      "paddingVertical": undefined,
      "position": undefined,
      "right": undefined,
      "top": undefined,
      "width": undefined,
      "zIndex": undefined,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    animated={false}
    collapsable={false}
    flex={0}
    focusable={true}
    height={89}
    justifyContentCenter={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    size="full"
    style={
      {
        "alignItems": undefined,
        "backgroundColor": undefined,
        "bottom": undefined,
        "flex": 0,
        "flexDirection": "column",
        "flexGrow": undefined,
        "gap": undefined,
        "height": 171,
        "justifyContent": "center",
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": 171,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "overflow": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
        "position": undefined,
        "right": undefined,
        "top": undefined,
        "width": "100%",
        "zIndex": undefined,
      }
    }
    testID="date-input-value-container"
  >
    <Text
      style={
        {
          "backgroundColor": undefined,
          "bottom": undefined,
          "color": "rgba(161, 173, 175, 1)",
          "flexGrow": undefined,
          "fontFamily": "Markpro-Medium",
          "fontSize": 18,
          "fontWeight": undefined,
          "gap": undefined,
          "height": undefined,
          "left": undefined,
          "letterSpacing": 2,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "maxHeight": undefined,
          "maxWidth": undefined,
          "minHeight": undefined,
          "minWidth": undefined,
          "overflow": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
          "position": undefined,
          "right": undefined,
          "textAlign": "auto",
          "top": undefined,
          "width": undefined,
          "zIndex": undefined,
        }
      }
    >
      Lütfen bir tarih seçiniz
    </Text>
  </View>
  <View
    absolute={true}
    bottom={16}
    right={0}
    style={
      {
        "alignItems": undefined,
        "backgroundColor": undefined,
        "bottom": 31,
        "flex": undefined,
        "flexDirection": "column",
        "flexGrow": undefined,
        "gap": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "left": undefined,
        "marginBottom": undefined,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "maxHeight": undefined,
        "maxWidth": undefined,
        "minHeight": undefined,
        "minWidth": undefined,
        "overflow": undefined,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "paddingVertical": undefined,
        "position": "absolute",
        "right": 0,
        "top": undefined,
        "width": undefined,
        "zIndex": undefined,
      }
    }
  >
    <View
      style={
        {
          "alignItems": undefined,
          "backgroundColor": undefined,
          "bottom": undefined,
          "flex": undefined,
          "flexDirection": "column",
          "flexGrow": undefined,
          "gap": undefined,
          "height": undefined,
          "justifyContent": undefined,
          "left": undefined,
          "marginBottom": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginTop": undefined,
          "maxHeight": undefined,
          "maxWidth": undefined,
          "minHeight": undefined,
          "minWidth": undefined,
          "overflow": undefined,
          "paddingBottom": undefined,
          "paddingHorizontal": undefined,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "paddingVertical": undefined,
          "position": undefined,
          "right": undefined,
          "top": undefined,
          "width": undefined,
          "zIndex": undefined,
        }
      }
    >
      <SvgMock
        fill="transparent"
        height={46}
        stroke="rgba(161, 173, 175, 1)"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={1}
        width={46}
      />
    </View>
  </View>
</View>
`;
